documentation/asciidoc/topics/ref_encryption_service_ca.adoc

```diff
@@ -0,0 +1,36 @@
+[id='encryption_with_service_ca-{context}']
+= Encryption with {openshift} Service Certificates
+{ispn_operator} automatically generates TLS certificates that are signed by the
+{openshift} service CA. {ispn_operator} then stores the certificates and keys
+in a secret so you can retrieve them and use with remote clients.
+
+If the {openshift} service CA is available, {ispn_operator} adds the following
+`spec.security.endpointEncryption` configuration to the **Infinispan** CR:
+
+[source,options="nowrap",subs=attributes+]
+----
+spec:
+  ...
+  security:
+    endpointEncryption:
+      type: service
```

Comment:
@rigazilla maybe we can change the endpointEncryption type to the CamelCase in order with other k8s and ocp? entities?

Reply:
We can use compare ignore case in code, but implement aligning with other k8s kind names.

Reply:
@dmvolod ignore case compare is here: https://github.com/infinispan/infinispan-operator/pull/479

Reply:
Cool, @rigazilla , thanks, will review

Reply:
@oraNod looks like we also need to fix endpointEncryption.type in proc_encrypting_endpoints_tls_secret.adoc to CamelCase

Reply:
@dmvolod thanks. I've changed it throughout so `type: CamelCase` - let me know if I've misunderstood the change.

Reply:
Sorry, @oraNod maybe I'm not so correctly wrote.
Camel Case style writing for type
https://en.m.wikipedia.org/wiki/Camel_case

Reply:
@dmvolod ok. isn't it correct that the first letter for any value of the `type` field needs to be caps? so `type: Value` not `type: value`

sorry if I'm making a meal of it at this stage. just getting a little confused bouncing between stuff.

Reply:
@oraNod , I'm not sure about all types, but for any `kind` type need to be started with uppercase

Reply:
thanks @dmvolod I think we should be good here in that case